packages/blob/src/put.ts

```diff
@@ -42,23 +42,25 @@ export function createPutMethod<
   getToken?: (pathname: string, options: T) => Promise<string>;
   extraChecks?: (options: T) => void;
 }) {
-  return async function put(
-    pathname: string,
-    body:
-      | string
-      | Readable
-      | Blob
-      | ArrayBuffer
-      | FormData
-      | ReadableStream
-      | File,
+  return async function put<TPath extends string>(
+    pathname: TPath,
+    body: TPath extends `${infer _rest}/`
+      ? undefined
+      :
+          | string
+          | Readable
+          | Blob
+          | ArrayBuffer
+          | FormData
+          | ReadableStream
+          | File,
     options?: T,
   ): Promise<PutBlobResult> {
     if (!pathname) {
       throw new BlobError('pathname is required');
     }
 
-    if (!body) {
+    if (!body && !pathname.endsWith('/')) {
```

Comment:
Let's add a comment that says this is the code that makes creating folders work.

Reply:
Should we also make it so that if you pass a pathname that ends with '/' and you also pass a body, then we warn about usage?

Reply:
we could even error early in here because the API returns an error as well, for all the JS people

Reply:
Yes we could do so indeed